test/scenarios/constantEffectInteractions/losingAndGainingKeywords.spec.ts

```diff
@@ -0,0 +1,56 @@
+describe('Losing and gaining keywords', function() {
+    integration(function(contextRef) {
+        describe('If a unit has a keyword given by a source, then loses that keyword due to an ability,', function() {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['specforce-soldier'],
+                        groundArena: ['wampa', 'crafty-smuggler']
+                    },
+                    player2: {
+                        groundArena: ['vigilant-honor-guards'],
+                        hand: ['protector', 'repair']
+                    }
+                });
+            });
+
+            it('it should be able to gain the keyword again from a different source', function () {
```

Comment:
Could we add a test of multiple instances of gain and then lose to make sure that it works for more than one iteration, that tends to be the issue with a lot of potential ways of implementing the blanking.

So like specifically, after the current test have another round of specforce soldier and then protector. Can see UniqueUnit.spec.ts for good patterns for how to handle duplicate cards